controller/lib/hal/hal_stm32.cpp

```diff
@@ -0,0 +1,1008 @@
+/* Copyright 2020, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+This file implements the HAL (Hardware Abstraction Layer) for the
+STM32L452 processor used on the controller.  Details of the processor's
+peripherals can be found in the reference manual for that processor:
+   https://www.st.com/resource/en/reference_manual/dm00151940-stm32l41xxx42xxx43xxx44xxx45xxx46xxx-advanced-armbased-32bit-mcus-stmicroelectronics.pdf
+
+Details specific to the ARM processor used in this chip can be found in
+the programmer's manual for the processor available here:
+   https://www.st.com/resource/en/programming_manual/dm00046982-stm32-cortexm4-mcus-and-mpus-programming-manual-stmicroelectronics.pdf
+
+*/
+
+#if defined(BARE_STM32)
+
+#include "hal.h"
+#include "hal_stm32.h"
+
+#define SYSTEM_STACK_SIZE       500
```

Comment:
Well, it's actually 2000 bytes, but you're right there's no reason to not make it larger.  I'll increase it.
I tend to keep a close eye on stack usage in the systems I work on because it's such a critical component to the systems correct functioning.  In my experience, a couple Kbytes is actually more then enough for a system like this, but that assumes that you don't go nuts allocating large structures to local memory.  That would be true for the code I write because I'm cognizant of the issue, but I can't guarantee that we'll all be aware of that.
I have a personal todo list of things to work on once this pull request is finished and I'll add some automatic stack checking to it.  That's an important thing to watch.